python/lsst/sdm/tools/_build_datalink_metadata.py

```diff
@@ -60,21 +60,21 @@ def filter_columns(table: dict[str, Any], filter_key: str) -> list[str]:
     """
     principal = []
     unknown_column_index = 100000000
-    for column in table["columns"]:
-        if column.get(filter_key):
-            column_index = column.get("tap:column_index", unknown_column_index)
+    for column in table.columns:
+        if getattr(column, filter_key.replace(":", "_")):  # e.g., tap:principal -> tap_principal
+            column_index = column.tap_column_index or unknown_column_index
             unknown_column_index += 1
-            principal.append((column["name"], column_index))
+            principal.append((column.name, column_index))
     return [c[0] for c in sorted(principal, key=lambda c: c[1])]
 
 
-def build_columns(felis: dict[str, Any], column_properties: list[str]) -> dict[str, dict[str, list[str]]]:
+def build_columns(schema: Schema, column_properties: list[str]) -> dict[str, dict[str, list[str]]]:
     """Find the list of tables with a particular Felis property.
 
     Parameters
     ----------
-    felis : Dict[`str`, Any]
-        The parsed Felis YAML.
+    schema: `~felis.Schema`
```

Comment:
```suggestion
    schema : `~felis.Schema`
```
The space is needed.

velin can fix these up automatically.